### DISCUSSION TITLE
Budibase died after hosting update

### DISCUSSION THREAD

**Discussion by R2bEEaton** [Budibase/budibase#11052] [Help]
Good morning,

We are running Budibase on-prem using the docker compose method via Budibase CLI. Yesterday I ran a routine update using the CLI `budi hosting --update` to update from `2.7.14` to the latest, `2.7.33`. Our CLI version is `9.7.2`. Yes, I realize this is old.

After running the update, which produced no errors, Budibase was showing a 502 Bad Gateway error through our nginx reverse proxy.

![image](https://github.com/Budibase/budibase/assets/34921506/cc7a183a-7d49-4fca-afb4-8e73d1e0175a)

I wasn't able to view http://localhost:10000/ due to not having a browser on the host machine, but I am seeing the following error when trying `wget localhost:10000`:

![image](https://github.com/Budibase/budibase/assets/34921506/1ec004fc-9f43-4086-a5e8-ea886273989c)

### What I tried after the problem happened:

1. Waited about 10 minutes for the error to go away, thinking it was similar to the temporary "Houston, we have a problem" error on startup
2. Restarted budibase `budi hosting --stop`, `budi hosting --start`
3. Killed all docker containers and started again
4. Viewed docker status `systemctl status docker` and saw some issue with the watchtower container about permissions, so I focued a lot of my time trying to fix that. Turns out it had the same thing before the problem started happening, so this is probably not it. I Googled the error and tried the suggestion, adding the root user to the `dockerroot` group which had no effect (I didn't think it would have).
5. Updated the Budibase CLI with npm
6. Tried to rollback update using #6353, not exactly sure if I did it right

For the moment, we resolved the problem by restoring a daily backup of the virtual server which was thankfully entirely dedicated to our Budibase instance. I had just made a backup of our main app right before the attempted update and all of our data are stored in a MSSQL database (#8331) now anyway. Suffice it to say that we got lucky. I have the broken server still, so I can provide any information needed.

Attached is my CentOS messages log from 14:50:00-15:00:00. The update took place at 14:53:12. Before uploading, I replaced `[0-9a-f]{8,}` with `OBFUSCATED` just in case, along with our URL. As I said, we're up and running with our backup restore, but it would be really nice to know what happened here. Thanks!
[Uploading 6-27-23-250messages.log…]()

**Comment by andz-bb** [comment]
hey @R2bEEaton, sorry to hear you had issues updating your instance. Thanks for your detailed report. I can take a look now to see if we can get to the bottom of this. Could you re-upload the log file please? I don't think it attached to the discussion properly. Also did you attempt to update again after restoring the server from the backup? Thanks

**Reply by R2bEEaton** [reply]
Hey @andz-bb thanks for getting back! Yes, sorry about that, here's the log. And no, we did not attempt to update again. We're understandably a little scared to do so. Thanks so much!

[6-27-23-250messages.log](https://github.com/Budibase/budibase/files/11895495/6-27-23-250messages.log)

**Reply by R2bEEaton** [reply]
@andz-bb No rush for us because we are content to stay on 2.7.14 for now, but just curious if there is any idea what went wrong. It was super shocking to break so dramatically because we had always had success with upgrading via the CLI.

**Reply by melohagan** [reply]
Hey @R2bEEaton I know it's been a while, but have you been able to update since?

**Reply by R2bEEaton** [reply]
Hi, since the update failed we have not had time to attempt another one. As mentioned before, we are content to remain on our current version, but if this is something you would like to explore with us, let me know!

**Reply by melohagan** [reply]
Thanks for getting back Ryan, if you do try updating again let me know how it goes!
